memo.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+require 'sinatra'
+require 'sinatra/reloader' if development?
+require 'json'
+require 'securerandom'
+
+get '/memos' do
+  @memos = read_memos
+
+  erb :index
+end
+
+get '/memos/new' do
+  erb :new
+end
+
+post '/memos' do
+  memos = read_memos
+  public_id = SecureRandom.uuid
+  memos[public_id] = { public_id:, title: params[:title], content: params[:content] }
+  write_memos(memos)
+
+  redirect "/memos/#{public_id}"
+end
+
+get '/memos/:public_id' do
+  @memo = find_memo(params[:public_id])
+
+  erb :show
+end
+
+get '/memos/:public_id/edit' do
+  @memo = find_memo(params[:public_id])
+
+  erb :edit
+end
+
+patch '/memos/:public_id' do
+  memos = read_memos
+  memos[params[:public_id].to_sym] = { public_id: params[:public_id], title: params[:title], content: params[:content] }
+  write_memos(memos)
+
+  redirect "/memos/#{params[:public_id]}"
+end
+
+delete '/memos/:public_id' do
+  memos = read_memos
+  memos.delete(params[:public_id].to_sym)
+  write_memos(memos)
+
+  redirect '/memos'
+end
+
+not_found do
+  '404 Not Found のページです'
+end
+
+def read_memos
+  File.open(db_path) do |file|
+    JSON.parse(file.read, symbolize_names: true)
+  end
+end
+
+def find_memo(public_id)
+  memos = read_memos
+  memo = memos[public_id.to_sym]
+  return not_found if memo.nil?
+
+  memo
```

Comment:
```suggestion
  memo or not_found
```

みたいに書くこともできそうです（ちょっとマニアックな書き方ですが）

Reply:
こういう書き方もあるんですね！
16f03c0 にて対応しました！